packages/block-editor/src/hooks/margin.js

```diff
@@ -167,33 +173,41 @@ export function MarginEdit( props ) {
 	} );
 }
 
-export function MarginVisualizer( { clientId, attributes } ) {
+export function MarginVisualizer( { clientId, attributes, isMouseOver } ) {
 	const margin = attributes?.style?.spacing?.margin;
-	const spacingSizes = useSetting( 'spacing.spacingSizes' );
 
 	const style = useMemo( () => {
 		const marginTop = margin?.top
-			? getCustomValueFromPreset( margin?.top, spacingSizes )
+			? getSpacingPresetCssVar( margin?.top )
 			: 0;
 		const marginRight = margin?.right
-			? getCustomValueFromPreset( margin?.right, spacingSizes )
+			? getSpacingPresetCssVar( margin?.right )
 			: 0;
 		const marginBottom = margin?.bottom
-			? getCustomValueFromPreset( margin?.bottom, spacingSizes )
+			? getSpacingPresetCssVar( margin?.bottom )
 			: 0;
 		const marginLeft = margin?.left
-			? getCustomValueFromPreset( margin?.left, spacingSizes )
+			? getSpacingPresetCssVar( margin?.left )
 			: 0;
 
 		return {
-			borderTopWidth: marginTop,
-			borderRightWidth: marginRight,
-			borderBottomWidth: marginBottom,
-			borderLeftWidth: marginLeft,
-			top: marginTop !== 0 ? `calc(${ marginTop } * -1)` : 0,
-			right: marginRight !== 0 ? `calc(${ marginRight } * -1)` : 0,
-			bottom: marginBottom !== 0 ? `calc(${ marginBottom } * -1)` : 0,
-			left: marginLeft !== 0 ? `calc(${ marginLeft } * -1)` : 0,
+			borderTopWidth: marginTop ? marginTop : 0,
+			borderRightWidth: marginRight ? marginRight : 0,
+			borderBottomWidth: marginBottom ? marginBottom : 0,
+			borderLeftWidth: marginLeft ? marginLeft : 0,
+			top: marginTop && marginTop !== 0 ? `calc(${ marginTop } * -1)` : 0,
+			right:
+				marginRight && marginRight !== 0
+					? `calc(${ marginRight } * -1)`
+					: 0,
+			bottom:
+				marginBottom && marginBottom !== 0
+					? `calc(${ marginBottom } * -1)`
+					: 0,
+			left:
+				marginLeft && marginLeft !== 0
+					? `calc(${ marginLeft } * -1)`
+					: 0,
```

Comment:
```suggestion
			borderTopWidth: marginTop,
			borderRightWidth: marginRight,
			borderBottomWidth: marginBottom,
			borderLeftWidth: marginLeft,
			top: marginTop ? `calc(${ marginTop } * -1)` : 0,
			right: marginRight ? `calc(${ marginRight } * -1)` : 0,
			bottom: marginBottom ? `calc(${ marginBottom } * -1)` : 0,
			left: marginLeft ? `calc(${ marginLeft } * -1)` : 0,
```

I think we have some unnecessary checks here that can be cleaned up a little.

Reply:
yep, that definitely seems to be a lot of unnecessary checks, what was I thinking 🤔  - thanks, fixed.